engine/access/rest/middleware/metrics.go

```diff
@@ -12,11 +12,13 @@ import (
 )
 
 func MetricsMiddleware(restCollector module.RestMetrics) mux.MiddlewareFunc {
-	metricsMiddleware := middleware.New(middleware.Config{Recorder: restCollector})
+	cfg := middleware.Config{Recorder: restCollector}
+	serviceID := cfg.Service
+	metricsMiddleware := middleware.New(cfg)
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 			// This is a custom metric being called on every http request
-			restCollector.AddTotalRequests(req.Context(), req.Method, req.URL.Path)
+			restCollector.AddTotalRequests(req.Context(), serviceID, req.Method)
```

Comment:
what would be the method? is it `GET/PUT/POST` etc?

Reply:
would it be more useful to have metrics on total requests to different path?
`Get /v1/events/*` 
`Get / v1/collections/*`

Reply:
tbh, if we're only collecting metrics for GET/POST, I don't think it's worth collecting the data.

How about adding something like this:
```golang
var routeUrlMap = map[string]string{}
var routeRE = regexp.MustCompile("/v1/([\\w]+)(/([\\w]+)(/([\\w]+))?)?")

func init() {
	for _, r := range rest.Routes {
		routeUrlMap[r.Pattern] = r.Name
	}
}

func URLToRoute(url string) (string, error) {
	normalized, err := normalizeURL(url)
	if err != nil {
		return "", err
	}

	name, ok := routeUrlMap[normalized]
	if !ok {
		return "", fmt.Errorf("invalid url")
	}
	return name, nil
}

func normalizeURL(url string) (string, error) {
	matches := routeRE.FindAllStringSubmatch(url, -1)
	if len(matches) != 1 || len(matches[0]) != 6 {
		return "", fmt.Errorf("invalid url")
	}

	// given a URL like
	//      /v1/blocks/1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef/payload
	// groups  [  1  ] [                                3                             ] [  5  ]
	// normalized form like /v1/blocks/{id}/payload

	parts := []string{matches[0][1]}

	//
	switch len(matches[0][3]) {
	case 0:
		// top level resource. e.g. /v1/blocks
	case 64:
		// id based resource. e.g. /v1/blocks/1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef
		parts = append(parts, "{id}")
	case 16:
		// address based resource. e.g. /v1/accounts/1234567890abcdef
		parts = append(parts, "{address}")
	default:
		// named resource. e.g. /v1/network/parameters
		parts = append(parts, matches[0][3])
	}

	if matches[0][5] != "" {
		parts = append(parts, matches[0][5])
	}

	return "/" + strings.Join(parts, "/"), nil
}
```

then using that as the handler name?

It's not very generic but probably good enough in the medium term.

Reply:
https://github.com/onflow/flow-go/compare/master...petera/add-rest-url-parser